### DISCUSSION TITLE
how to add LDAP groups in docker-compose setup

### DISCUSSION THREAD

**Discussion by zisuu** [WeblateOrg/weblate#8978] [Q&A]
Hello Weblater's

I love this Project, keep going on, you're doing great! However I have a Problem and I guess its just me not getting how it works:

I've running Weblate Instance with docker-compose (https://github.com/WeblateOrg/docker-compose). So far I went well with using just the LDAP Backend and adding the users manually if they did not login before. But now I need to onboard a Project with 30+ LDAP Users, so now is a good moment to check if and how I could just use the LDAP Group (would just be one which contains all 30+ users) instead of bulk adding each user.

I found this related to AD Groups: https://docs.weblate.org/de/latest/admin/auth.html#active-directory-integration

but I don't understand this. Can someone help me with the questions I have:

1. What should I do with this Python script? (https://docs.weblate.org/de/latest/admin/auth.html#active-directory-integration) I'm running a docker-compose Weblate Instance, how will that be presistent if I just run some Python script? Whats the correct way to apply this python script without breaking the existing LDAP Config I have in the environment.env:
```
# LDAP Auth
WEBLATE_AUTH_LDAP_BIND_DN="CN=blabla,OU=services,OU=users,DC=example,DC=com"
WEBLATE_AUTH_LDAP_BIND_PASSWORD={{ vault_weblate.secret.WEBLATE_AUTH_LDAP_BIND_PASSWORD }}
WEBLATE_AUTH_LDAP_SERVER_URI=ldaps://server.example.com:636
WEBLATE_AUTH_LDAP_USER_ATTR_MAP=full_name:name,email:mail
WEBLATE_AUTH_LDAP_USER_SEARCH="OU=users,DC=example,DC=com"
WEBLATE_AUTH_LDAP_USER_SEARCH_FILTER="(sAMAccountName=%(user)s)"
```

2. How can I add only the AD Groups I need, without enabling AUTH_LDAP_MIRROR_GROUPS? I can only find the Option to add Teams in the UI. Do I need to create the Groups in the Django Admin Panel?

3. Are "AD Groups" equal to " Weblate Teams"? Or what would be the Workflow to create a new Weblate Project and only allow the Users of the AD-Group "XY" to Administrate the new Project? I only see the Option to add Users to a Projects "Administration Team", but no Option to add AD-Groups?

Thanks a lot for your support, support gratefully welcomed ;-)

Cheers

**Comment by zisuu** [comment]
anyone?

**Comment by nijel** [comment]
Any Python configuration can be used in Docker as well, see https://docs.weblate.org/en/latest/admin/install/docker.html#configuration-beyond-environment-variables. The Docker environment variables only allow limited configuration and cannot fully mirror Python capabilities.

I'm uncertain if [AUTH_LDAP_MIRROR_GROUPS](https://django-auth-ldap.readthedocs.io/en/latest/reference.html#std:setting-AUTH_LDAP_MIRROR_GROUPS) or [AUTH_LDAP_FIND_GROUP_PERMS](https://django-auth-ldap.readthedocs.io/en/latest/reference.html#std:setting-AUTH_LDAP_FIND_GROUP_PERMS) will work with Weblate (it doesn't use Django permissions).
